core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -341,6 +323,45 @@ public KeyStore getPrimaryKeyStore() throws Exception {
         }
     }
 
+    /**
+     * Load the requested tenant keystore from registry.
+     *
+     * @param keyStoreName  Name of the Tenant KeyStore.
+     * @return Key store object.
+     * @throws Exception    Exception if failed to retrive the given keystore from registry.
+     */
+    private KeyStore getTenantKeyStore(String keyStoreName) throws Exception {
```

Comment:
Suggest to throw a specifically defined Exception rather than just throwing an `Exception`

Reply:
Same scenario as https://github.com/wso2/carbon-kernel/pull/4025#discussion_r1687729301